auth/handler_utils.go

```
@@ -0,0 +1,141 @@
+// from context if set (e.g. by gRPC gateway).
+func GetPublicURL(ctx context.Context, req *http.Request, cfg *config.Config) *url.URL {
+	u := FirstURL(URLFromRequest(req), URLFromContext(ctx))
+	var hostMatching *url.URL
+	var hostAndPortMatching *url.URL
+	for _, authorized := range cfg.AuthorizedURIs {
+		if u == nil {
+			return &authorized.URL
+		}
+
+		if u.Hostname() == authorized.Hostname() {
+			hostMatching = &authorized.URL
```

golangci-lint (1.64.8)

[medium] G601: Implicit memory aliasing in for loop.

(gosec)